lib/src/signed_video_h26x_auth.c

```diff
@@ -879,6 +894,8 @@ prepare_for_validation(signed_video_t *self)
       if (self->gop_info->signature_hash_type == DOCUMENT_HASH) {
         memcpy(verify_data->hash, sei->hash, hash_size);
       }
+    } else if (sei && sei->has_been_decoded) {
```

Comment:
It is starting to become too many if-statements in this function. What is done in `detect_sei_loss_and_sync_gop` below, is also done in `decode_sei_data`. It should be possible to call the same function for both cases instead of having code duplications wrapped in various if statements. This will make the code more readable and less error prone.